internal/tofu/eval_variable.go

```diff
@@ -477,3 +477,29 @@ You can correct this by removing references to sensitive values, or by carefully
 		FailureMessage: errorMessage,
 	}, diags
 }
+
+// evalVariableDeprecation checks if a variable is deprecated and if so it returns a warning diagnostic to be shown to the user
+func evalVariableDeprecation(addr addrs.AbsInputVariableInstance, config *configs.Variable, expr hcl.Expression, ctx EvalContext) tfdiags.Diagnostics {
+	if config.Deprecated == "" {
+		log.Printf("[TRACE] evalVariableDeprecation: variable %s is having no deprecation configured", addr)
```

Comment:
Nit: "is having no" is not conventional English.  "does not have" is more standard here.